pkg/cmd/roachtest/tests/versionupgrade.go

```diff
@@ -92,6 +92,7 @@ func runVersionUpgrade(ctx context.Context, t test.Test, c cluster.Cluster) {
 	// the 20.2 release, this test will fail because it is missing a fixture for
 	// 20.1; run the test (on 20.1) with the bool flipped to create the fixture.
 	// Check it in (instructions will be logged below) and off we go.
+	// TODO(celia) - make sure this is false before merging back!
```

Comment:
This comment can be removed.

I think the problem with the fixture generation is that `makeFixtureVersion` below is set to a value that assumes that 21.2.0 exists already, but it does not. You need to download beta1 to your local machine, point `roachtest` at it with the `--cockroach` flag, and run with `makeFixtureVersion = ""`. Then we will start your local binary (as opposed to trying to download nonexistent 21.2.0), extract the version from that, and learn the predecessor from that and then make the fixture. That is what the comment on line 97-114 tries to explain, if you read that but did not draw those conclusions please reword it so that next time it will be clearer, thank you!

It's also possible that you can make it work with just using `21.2.0-beta.1` for `makeFixtureVersion` which would be easier, but I don't think it will - might be worth a shot though since it's so easy to try.

Reply:
I'm still getting an error -- not sure if I'm still unclear on the exact steps to run or not. So I'll add them below. But this is the error I'm now seeing:

```
--- FAIL: acceptance/version-upgrade (139.78s)
test artifacts and logs in: artifacts/acceptance/version-upgrade/run_1
	versionupgrade.go:480,versionupgrade.go:214,versionupgrade.go:644,versionupgrade.go:116,acceptance.go:54,acceptance.go:104,
test_runner.go:777: 1: expected version 21.1-1162, got 21.1-1124

```

Here's the steps I'm running:

```
## Download beta.1
curl https://binaries.cockroachdb.com/cockroach-v21.2.0-beta.1.darwin-10.9-amd64.tgz \
| tar -xJ && cp -i cockroach-v21.2.0-beta.1.darwin-10.9-amd64/cockroach /usr/local/bin/

## Verify version

Celia’s-MacBook-Pro:cockroach celiala$ /usr/local/bin/cockroach version
Build Tag:        v21.2.0-beta.1
Build Time:       2021/09/24 15:24:05
Distribution:     CCL
Platform:         darwin amd64 (x86_64-apple-darwin19)
Go Version:       go1.16.6
C Compiler:       Clang 10.0.0
Build Commit ID:  215694e4ca58b24701d5621606aa1bfbd5783eb9
Build Type:       release
Celia’s-MacBook-Pro:cockroach celiala$

## Make clean builds of CRDB, roachtest, workload, roachprod

make build # generate the full crdb binary, start at master
make bin/roachtest
make bin/workload
make bin/roachprod

## Clear out roachprod remnants, if any

bin/roachprod wipe local # just to clear out remnants, if any
bin/roachprod destroy local

## Run roachtest against beta.1 binary

bin/roachtest --local run acceptance/version-upgrade \
--debug --cockroach /usr/local/bin/cockroach

```

Reply:
Hmm, these steps look right. What if you bump the 30s here to something like 5 minutes? I have noticed elsewhere that the long-running migrations might take quite long to run.

https://github.com/cockroachdb/cockroach/blob/fbfcded982530a9b37af3f2c843da2769f927025/pkg/cmd/roachtest/tests/versionupgrade.go#L470